File: app/src/main/java/com/pavel/a692group/room/dao/UserDao.java
package com.pavel.a692group.room.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.pavel.a692group.room.entity.User;

import java.util.List;

import io.reactivex.Flowable;

/**
 * Created by Pavel Mazhnik on 02.03.19.
 * Description:
 */

@Dao
public interface UserDao {

    @Query("SELECT * FROM " + User.TABLE)
    Flowable<List<User>> getAll();

    @Query("SELECT * FROM " + User.TABLE + " WHERE " + User.COLUMN_ID + " = :id")
    Flowable<List<User>> getById(long id);

    @Query("SELECT DISTINCT " + User.COLUMN_GROUP + " FROM " + User.TABLE)
    Flowable<List<String>> getAllGroup();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(User user);

    /*@Update
    int update(User user);*/

    @Delete
    void delete(User user);
}

File: app/src/main/java/com/pavel/a692group/room/dao/UserDao.kt
package com.pavel.a692group.room.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.pavel.a692group.room.entity.User

/**
 * Created by Pavel Mazhnik on 02.03.19.
 * Description:
 */
@Dao
interface UserDao {
    @get:Query("SELECT * FROM " + User.TABLE)
    val all: LiveData<List<User?>?>?

    @Query("SELECT * FROM " + User.TABLE + " WHERE " + User.COLUMN_ID + " = :id")
    fun getById(id: Long): LiveData<List<User?>?>?

    @get:Query("SELECT DISTINCT " + User.COLUMN_GROUP + " FROM " + User.TABLE)
    val allGroups: LiveData<List<String?>?>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(user: User?)

    /*@Update
    int update(User user);*/
    @Delete
    suspend fun delete(user: User?)
}

